framework/src/modules/random/schemas.ts

```diff
@@ -122,26 +131,35 @@ export const getHashOnionUsageResponse = {
 };
 
 export interface SetHashOnionUsageRequest extends AddressRequest {
-	height: number;
-	count: number;
+	usedHashOnions: UsedHashOnion[];
 }
 
 export const setHashOnionUsageRequest = {
 	$id: 'lisk/random/setHashOnionUsageRequest',
 	type: 'object',
-	required: ['address', 'count', 'height'],
+	required: ['address', 'usedHashOnions'],
 	properties: {
 		address: {
 			type: 'string',
 			format: 'lisk32',
 		},
-		count: {
-			type: 'integer',
-			minimum: 1,
-		},
-		height: {
-			type: 'integer',
-			format: 'uint32',
+		usedHashOnions: {
+			type: 'array',
+			items: {
+				type: 'object',
+				required: ['count', 'height'],
+				properties: {
+					count: {
+						type: 'integer',
+						format: 'uint32',
+						minimum: 1,
```

Comment:
isn't first count stored `0`?

Reply:
can you elaborate on what you mean by "first count" ? I think count `0` as input doesn't make sense for the endpoint `setHashOnionUsage`. Also as per my understanding count is only set to `0` internally in the `setLatest` function to reset the counter in case of non-matching `usedHashOnion` data.

Reply:
After the first usage count should be 1, why do you think it is 0 ?

Reply:
if it's first time, it will go in to this condition https://github.com/LiskHQ/lisk-sdk/blob/1e4eb7293d7cb2c12f06593533e338856649cb9d/framework/src/modules/random/module.ts#L225-L229 and count: 0 will be saved?

Reply:
I am not sure when exactly the above logic gets executed after the node startup once the seed is set but for the first time node forges a block once the seed is set `getHashOnionUsage` returns an output with `count: 1`. For eg.  `"height": 10, "count": 1,`.  Or do you mean some other scenario ?